### PR TITLE
Unit tests and fixes for R/models.R

### PR DESCRIPTION
**Pull request by nikolase90**
Summarized
- Changes cases of `'something'` to `"something"` in `R/models.R`
- Modifies `predict_model.*` `R/models.R` so that it handles both data.frame and matrix as input.
- Adds check for `ranger`. If you're passing a classification model it needs to be a probability forest (if that is not the case you'll only get predicted classes and not probabilities). 
- Adds tests in `tests/testthat/test-models.R`

### CODE REVIEW

File: R/models.R [hunk 1]
```diff
@@ -142,8 +154,20 @@ model_type.glm <- function(x) {
...
 #' @export
 model_type.ranger <- function(x) {
+
+  if (x$treetype == "Classification") {
+    stop(
+      paste0(
+        "\n",
+        "We currently don't support standard classification using ranger.\n",
+        "You'll need to grow a probability forest by setting probability = TRUE\n",
+        "in ranger::ranger(), while training a classification model using the ranger package."
```
[the remark this answers is not in the document]
Comment (nikolase90) [reply R/models.R]: Fixed with 764a720.

File: R/models.R [hunk 2]
```diff
@@ -73,7 +74,18 @@ predict_model.glm <- function(x, newdata) {
...
+    stop("The ranger package is required for predicting ranger models")
+  }
+
+  if (x$treetype == "Classification") {
+    stop(
+      paste0(
+        "\n",
+        "We currently don't support standard classification using ranger.\n",
+        "You'll need to grow a probability forest by setting probability = TRUE\n",
+        "in ranger::ranger(), while training a classification model using the ranger package."
```
[the remark this answers is not in the document]
Comment (nikolase90) [reply R/models.R]: Yeah, that's true, but I try (my best) to make the unit tests somewhat independent of how we run the code as a whole. Changed it to d108711.

**Comment by nikolase90** [comment]
> This is good to have.
> Since we only support binary classification and not multiclass classification, we should probably have checks for that too. I.e. a stop in either the model_type (preferrably) or predict_model if a multiclass model is provided. Can you fix that?

:+1:

**Comment by nikolase90** [comment]
@martinju Updated with tests for multi-classification.

File: R/models.R [hunk 3]
```diff
@@ -142,11 +149,33 @@ model_type.glm <- function(x) {
...
+        "\n",
+        "We currently don't support standard classification, which predicts the class directly.\n",
+        "To train a ranger model predicting the class probabilities, you'll need to grow a\n",
+        "probability forest by setting probability = TRUE in ranger::ranger()."
+      )
+    )
+  }
+
+  if (x$treetype == "Probability estimation") {
+    if (length(x$fores$levels) == 2) {
```
Comment (martinju) [inline R/models.R]: Typo: fores -> forest

File: R/models.R [hunk 4]
```diff
@@ -142,11 +149,33 @@ model_type.glm <- function(x) {
...
+      "classification"
+    } else {
+      stop(
+        paste0(
+          "\n",
+          "We currently don't support multi-classification using ranger, i.e.\n",
+          "where length(model$fores$levels) is greater than 2."
```
Comment (martinju) [inline R/models.R]: Typo: fores -> forest

**Comment by nikolase90** [comment]
Fixed typos with 7ed8625.

**Pull request merged**